pkg/k8sutil/namespace.go

```diff
@@ -32,10 +32,23 @@ func EnsureNamespace(client kubernetes.Interface, namespace string) error {
 
 // EnsureNamespaceWithLabel creates a namespace with optional labels
 func EnsureNamespaceWithLabel(client kubernetes.Interface, namespace string, labels map[string]string) error {
+
+	// add label to mark the namespace that it was created by Pipeline
+	mergedLabels := map[string]string{}
+	if labels != nil {
+		for k, v := range labels {
+			mergedLabels[k] = v
+		}
+	}
+
+	if _, ok := mergedLabels["owner"]; !ok {
```

Comment:
you could put the default entries in the map first, than let entries from the incoming map overwrite them
```go
mergedLabels := map[string]string{
	"owner": "pipeline",
}
for k, v := range labels {
	mergedLabels[k] = v
}
```